nncf/openvino/statistics/collectors.py

```diff
@@ -69,30 +70,36 @@ def max(x1: NNCFTensor, x2: NNCFTensor) -> NNCFTensor:
         return OVNNCFTensor(np.maximum(x1.tensor, x2.tensor))
 
     @staticmethod
-    def mean(x: NNCFTensor, axis: Union[int, tuple], keepdims: bool = False) -> NNCFTensor:
+    def mean(x: NNCFTensor, axis: Union[int, Tuple], keepdims: bool = False) -> NNCFTensor:
         return OVNNCFTensor(np.mean(x.tensor, axis=axis, keepdims=keepdims))
 
     @staticmethod
-    def median(x: NNCFTensor, axis: Union[int, tuple, list], keepdims: bool = False) -> NNCFTensor:
+    def median(x: NNCFTensor, axis: Union[int, Tuple, list], keepdims: bool = False) -> NNCFTensor:
         return OVNNCFTensor(np.median(x.tensor, axis=axis, keepdims=keepdims))
 
     @classmethod
     def masked_mean(
-        cls, x: NNCFTensor, axis: Optional[Union[int, tuple, list]], mask: Optional[NNCFTensor], keepdims: bool = False
+        cls, x: NNCFTensor, axis: Optional[Union[int, Tuple, list]], mask: Optional[NNCFTensor], keepdims: bool = False
     ) -> NNCFTensor:
         if mask is None:
             return cls.mean(x, axis=axis, keepdims=keepdims)
         masked_x = np.ma.array(x.tensor, mask=mask.tensor)
-        return OVNNCFTensor(np.ma.mean(masked_x, axis=axis, keepdims=False).data)
+        result = np.ma.mean(masked_x, axis=axis, keepdims=keepdims)
+        if isinstance(result, np.ma.MaskedArray):
+            return OVNNCFTensor(result.data)
+        return OVNNCFTensor(result)
```

Comment:
`OVNNCFTensor` should always contain either `np.ndarray` tensor or numpy scalar . Masked operations are using `np.ma.MaskedArray` inside as this is a convenient way to work with masked arrays in numpy. For some reasons result of mean operation is a numpy scalar in case result is 0 dimensions tensor (scalar) and MaskedArray otherwise. In case a MaskedArray returned by the operation it is casted to `np.ndarray` by taking `.data` attribute from it